### DISCUSSION TITLE
Understanding positioning

### DISCUSSION THREAD

**Discussion by ThonAfterPy** [zauberzeug/nicegui#957] [Q&A] [answered]
### Question

Hello. I am very new to front-end development, my only experience I ever had developing UI's is with game development. In the game engines, I could set an elements position by either: a decimal percentage of the screen's width/height (I.E. x=.5 and y=.5 would be in the center of the screen) and/or by an offset of pixels (I.E. x=50 y=50), as well as changing an elements anchor point. Now, I have come to realize it is a lot more complex to understand in web-development.
I have looked at many resources such as the Tailwind and Quasar classes, but I could not find the answer to what I am looking for. I'm sure it is in their somewhere, but I am lost to what to even look for.
So, basically, just so I can get my footing somewhere and go from there, how would I position an element in the way I used to do it described above? As in, how would I get an element to be, in this case, 50 pixels below and 50 pixels to the right of the center of the screen (X=0.5, Y=0.5). I have looked into using justify and center to center elements, which does work, but it lacks the ability to change the position to be at 51% of the screens width and 51% of the screens height for example.

**Comment by falkoschindler** [comment, marked as answer]
To position an element absolutely on the screen, you might be better off with plain CSS instead of Tailwind classes (even though almost everything is possible with Tailwind as well):

```py
ui.label('Hello CSS!').classes('border p-4') \
    .style('position: absolute; top: 50%; left: 50%; transform: translate(-50%, -50%);')
```

Let me explain:
- `position: absolute` specifies the positioning method.
- `top: 50%; left: 50%` specifies the position.
- `transform: translate(-50%, -50%)` moves the label such that its center is at the position defined with `top` and `left`.

With Tailwind the code would look like this:
```py
ui.label('Hello Tailwind!') \
    .classes('border p-4 absolute top-[50%] left-[50%] translate-x-[-50%] translate-y-[-50%]')
```
